File: src/AnalyticsService.Database/Api/Extensions.cs
using AnalyticsService.Database.Repositories;
using Microsoft.Extensions.DependencyInjection;

namespace AnalyticsService.Database.Api;

public static class Extensions
{
    /// <summary>
    /// An extension method for registering/adding repositories as services.
    /// </summary>
    public static IServiceCollection AddRepositories(this IServiceCollection services)
    {
        return services
            .AddSingleton<IDlqRepository, DlqRepository>();
    }
}

File: src/AnalyticsService.Database/Api/Extensions.fs
[<Sealed>]
module AnalyticsService.Database.Api.Extensions

open AnalyticsService.Database.Repositories
open Microsoft.Extensions.DependencyInjection

type IServiceCollection with
    /// An extension method for registering/adding repositories as services.
    member services.AddRepositories () =
        services
            .AddSingleton<IDlqRepository, DlqRepository>()
            .AddSingleton<IBatchStatRepository, BatchStatRepository>()
